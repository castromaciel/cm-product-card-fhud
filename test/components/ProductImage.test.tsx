import React from 'react'
import { product2 } from '../data/data'
import renderer from 'react-test-renderer';
import { ProductImage } from '../../src/components/ProductImage';
import { ProductCard } from '../../src/components/ProductCard';

describe('ProductImage', () => {
  test('Debe de mostrar el componente correctamente con la imagen personalizada', () => {

    const wrapper = renderer.create(
      <ProductImage img={'https://hola.com.ar'} title={'Title'}/>
    )

    expect( wrapper.toJSON() ).toMatchSnapshot()
  })

  test('Debe de mostrar el componente correctamente con la imagen del producto', () => {

    const wrapper = renderer.create(
      <ProductCard product={product2}>
        {
          () => (
            <ProductImage />
          )
        }
      </ProductCard>
    )

    expect( wrapper.toJSON() ).toMatchSnapshot()
    
  })

})