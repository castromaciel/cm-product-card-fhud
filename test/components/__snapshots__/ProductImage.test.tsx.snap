// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe de mostrar el componente correctamente con la imagen del producto 1`] = `
<article
  className=" productCard undefined"
>
  <img
    alt="Coffe Mug - Card"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</article>
`;

exports[`ProductImage Debe de mostrar el componente correctamente con la imagen personalizada 1`] = `
<img
  alt="Title"
  className="productImg undefined"
  src="https://hola.com.ar"
/>
`;
